specification/redisenterprise/resource-manager/Microsoft.Cache/preview/2023-03-01-preview/redisenterprise.json

```diff
@@ -0,0 +1,2046 @@
+          "items": {
+            "$ref": "#/definitions/RegionSkuDetail"
+          },
+          "description": "List of Sku Detail",
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "LocationInfo": {
+      "type": "object",
+      "description": "Information about location (for example: features that it supports)",
+      "properties": {
+        "location": {
+          "type": "string",
+          "description": "Location name"
+        },
+        "capabilities": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Capability"
+          },
+          "description": "List of capabilities",
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "SkuDetail": {
+      "type": "object",
+      "title": "Sku Detail",
+      "description": "Information about Sku",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The type of RedisEnterprise cluster to deploy. Possible values: (Enterprise_E10, EnterpriseFlash_F300 etc.)",
+          "enum": [
+            "Enterprise_E10",
+            "Enterprise_E20",
+            "Enterprise_E50",
+            "Enterprise_E100",
+            "EnterpriseFlash_F300",
+            "EnterpriseFlash_F700",
+            "EnterpriseFlash_F1500"
+          ],
+          "x-ms-enum": {
+            "name": "name",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "Capability": {
+      "type": "object",
+      "title": "Capability",
+      "description": "Information about the features the location supports",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Feature name"
+        },
+        "value": {
+          "type": "boolean",
+          "description": "Indicates whether feature is supported or not"
+        }
+      }
+    },
+    "RegionSkuDetail": {
+      "title": "Region Sku Detail",
+      "description": "Details about the location requested and the available skus in the location",
+      "type": "object",
+      "properties": {
+        "resourceType": {
+          "type": "string",
+          "description": "Resource type which has the SKU, such as Microsoft.Cache/redisEnterprise"
+        },
+        "locationInfo": {
+          "$ref": "#/definitions/LocationInfo",
+          "description": "Details about location and its capabilities"
+        },
+        "skuDetails": {
+          "$ref": "#/definitions/SkuDetail",
+          "description": "Details about available skus"
+        }
+      }
+    },
+    "FlushParameters": {
+      "title": "Flush all the keys from the current and all its linked databases",
+      "type": "object",
+      "properties": {
+        "ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The resource IDs of the database resources to be flushed"
+        }
+      },
+      "required": [
+        "ids"
```

Comment:
> "ids" [](http://example.com/codeflow?start=7&length=6)

Could this parameter be made optional, and then the same API be used for flushing non-georeplicated databases also?

Reply:
I am not sure of making this optional, as now we say it is optional and throw an error when the ids are not specified.

One more question is, if we make it required now and change it to optional later when supporting non-geo replicated , will that make a breaking change?

Reply:
Mainly that is non-breaking, but IIUC in the details of things like generated SDK code it might possibly look breaking.
Also, as another scenario, can it be totally optional for a CRDB database which isn't linked to any other region yet?

Reply:
What i think is, it would be better to keep it required , but still accept an empty array.
With this, 
1. we can still use this for non-georeplicated caches in the future
2. works for a CRDB database which is not linked to any other region yet.
3. if the CRDB database is linked to other regions, then those should be specified which we will be verifying whether all the linked databases are specified or not.

Reply:
Yes but... its not user-friendly to require passing an empty array.

Reply:
made them optional